added bogosort algorithem

mendlero opened the pull request:
added bogosort algorithem

appgurueu reviewed (changes requested):
* The Fisher-Yates `shuffle` and `isSorted` should get their own files (and tests); they are much more useful than bogosort. This will trivialize the bogosort.
* Don't use `var`, use `let` and `const` instead.
* The complexity analysis is wrong. The best case complexity is O(n), since a single `isSorted` check takes O(n). The (expected) average complexity is not "unbounded"; yes, it can in theory fail to find the correct permutation infinitely often, but the probability for that is 0: Each iteration has a probability of 1/n! of succeeding. You obtain an expected O(n!n) time complexity using the geometric distribution since each iteration takes O(n) time.
* You need to add tests. Please try to do this without simply duplicating tests.

mendlero commented:
> * The Fisher-Yates `shuffle` and `isSorted` should get their own files (and tests); they are much more useful than bogosort. This will trivialize the bogosort.
> * Don't use `var`, use `let` and `const` instead.
> * The complexity analysis is wrong. The best case complexity is O(n), since a single `isSorted` check takes O(n). The (expected) average complexity is not "unbounded"; yes, it can in theory fail to find the correct permutation infinitely often, but the probability for that is 0: Each iteration has a probability of 1/n! of succeeding. You obtain an expected O(n!n) time complexity using the geometric distribution since each iteration takes O(n) time.
> * You need to add tests. Please try to do this without simply duplicating tests.

Thanks for reviewing so quickly, I refactored the code, added tests and change the time complexity in the latest commits.

appgurueu commented on other/shuffle_array.ts:
```diff
@@ -0,0 +1,14 @@
+ * @function shuffleArray
+ * @description shuffles an array.
+ * @param {number[]} arr - array to shuffle
+ * @returns {void} - the function does not return anything
```
This is self explanatory

appgurueu commented on sorts/test/bogo_sort.test.ts:
```diff
@@ -0,0 +1,16 @@
+import { bogoSort } from '../bogo_sort';
+
+describe('BubbleSort', () => {
+	it('should return the correct values for test case 1', () => {
```
These test case names aren't useful. See [the contribution guidelines](https://github.com/TheAlgorithms/TypeScript/blob/master/CONTRIBUTING.md#writing-good-tests). Please simply remove the `it` blocks or use `each`.

mendlero replied:
Thanks for pointing it out, I changed the tests to use test.each

appgurueu commented on sorts/bogo_sort.ts:
```diff
@@ -1,11 +1,14 @@
+ *                      The best case occurs when the array is already sorted.
  *      Worst case  -   unbounded
  *                      The worst case occurs when the suffeles never make the array sorted.
  *      Average case -  unbounded
```
You haven't fixed the analysis of the average case.

mendlero replied:
please take a look at the last commit
